protocol/extend.go

```diff
@@ -1,13 +1,39 @@
 package protocol
 
-import "github.com/jackc/pgx/pgproto3"
+import (
+	"fmt"
+	"github.com/jackc/pgx/pgio"
+	"github.com/jackc/pgx/pgproto3"
+	nodes "github.com/lfittl/pg_query_go/nodes"
+)
 
 // ParseComplete is sent when backend parsed a prepared statement successfully
 var ParseComplete = []byte{'1', 0, 0, 0, 4}
 
 // BindComplete is sent when backend prepared a portal and finished planning the query
 var BindComplete = []byte{'2', 0, 0, 0, 4}
 
+// ParameterDescription is sent when backend received Describe message from frontend
+// with ObjectType = 'S' - requesting to describe prepared statement with a provided name
+func ParameterDescription(ps *nodes.PrepareStmt) (Message, error) {
+	res := []byte{'t'}
+	sp := len(res)
+	res = pgio.AppendInt32(res, -1)
+
+	res = pgio.AppendUint16(res, uint16(len(ps.Argtypes.Items)))
+	for _, v := range ps.Argtypes.Items {
+		p, ok := v.(nodes.TypeName)
```

Comment:
Every item must be a `TypeName` according to the docs:
```
/* Types of parameters (List of TypeName) */
```
If you didn't find a specific case where this comment is wrong, you can safely assume `TypeName`.